tools/cli/commands.go

```diff
@@ -1148,12 +1142,20 @@ func getCliIdentity() string {
 	return fmt.Sprintf("cadence-cli@%s", hostName)
 }
 
-func newContext() (context.Context, context.CancelFunc) {
-	return context.WithTimeout(context.Background(), time.Second*5)
+func newContext(c *cli.Context) (context.Context, context.CancelFunc) {
+	contextTimeout := defaultContextTimeout
+	if c.GlobalInt(FlagContextTimeout) > 0 {
+		contextTimeout = time.Duration(c.GlobalInt(FlagContextTimeout)) * time.Second
+	}
+	return context.WithTimeout(context.Background(), contextTimeout)
 }
 
-func newContextForLongPoll(timeout time.Duration) (context.Context, context.CancelFunc) {
-	return context.WithTimeout(context.Background(), timeout)
+func newContextForLongPoll(c *cli.Context) (context.Context, context.CancelFunc) {
+	contextTimeout := defaultContextTimeoutForLongPoll
+	if c.GlobalInt(FlagContextTimeout) > 0 {
```

Comment:
I think because you set default value for FlagContextTimeout, this condition will always be true and lead to unexpected timeout for longpoll. Maybe use c.GlobalIsSet or remove default flag value.

Reply:
You are right, I missed that. I will use GlobalIsSet